packages/flutter/lib/src/widgets/asset_vendor.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+
+/// Establishes an asset resolution strategy for its descendants.
+///
+/// An AssetVendor establishes a [ResolvingAssetBundle] as the default bundle
+/// for its subtree. The resolution logic is provided by an [AssetResolver]
+/// specified via its [AssetResolverBuilder].
+class AssetVendor extends StatefulComponent {
+  AssetVendor({
+    Key key,
+    this.bundle,
+    this.builder,
+    this.child
+  }) : super(key: key);
+
+  final AssetBundle bundle;
+  final AssetResolverBuilder builder;
+  final Widget child;
+
+  _AssetVendorState createState() => new _AssetVendorState();
+}
+
+class _AssetVendorState extends State<AssetVendor> {
+
+  ResolvingAssetBundle bundle;
+
+  void initState() {
+    super.initState();
+    this.bundle = new ResolvingAssetBundle(
+      bundle: config.bundle,
+      resolver: config.builder.buildResolver()
```

Comment:
I thought this was the convention.  State<T> says of initState:

"Override this function to perform initialization that depends on the location at which this object was inserted into the tree or on the widget configuration object."

I'm depending on the widget configuration object here, so I thought this was the right place to do the initialization.